task_processing/executors/mesos_executor.py

```diff
@@ -1,21 +1,19 @@
-# -*- coding: utf-8 -*-
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
 import threading
-from collections import namedtuple
 
 import mesos.native
-from execution_framework import ExecutionFramework
 from mesos.interface import mesos_pb2
-TaskConfig = namedtuple(
-    'TaskConfig',
-    ['image', 'cmd', 'cpus', 'mem', 'disk', 'volumes', 'ports'],
-)
 
+from task_processing.executors.task_executor import Asyncable
+from task_processing.executors.task_executor import Promiseable
+from task_processing.executors.task_executor import Subscribable
+from task_processing.executors.task_executor import TaskExecutor
+from task_processing.task_processing.execution_framework import ExecutionFramework
 
-class TaskExecutor(object):
-    """ Interface for task execution."""
+
+class MesosExecutor(TaskExecutor, Promiseable, Asyncable, Subscribable):
```

Comment:
notes after our chat irl: sync, async, promise and subscription are all functions of mesos driver that provide some concurrency interface.

    executor = MesosExecutor(concurrency_interface=Synchronous(), credentials, ...)
    executor.run(task_config)

    # for subscription interface
    executor = MesosExecutor(concurrency_interface=Subscription(my_queue), ...)
    my_queue.push(task_config)
    my_queue.pop() # -> status updates

MesosExecutor is then a class that wires together ExecutionFramework, mesos driver and concurrency interface.